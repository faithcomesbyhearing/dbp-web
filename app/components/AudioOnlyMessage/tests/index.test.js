import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import AudioOnlyMessage from '..';

Enzyme.configure({ adapter: new Adapter() });

/* eslint-disable react/prop-types */
jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));
/* eslint-enable react/prop-types */

const book = 'Romans';
const chapter = 8;

describe('AudioOnlyMessage Component', () => {
	it('Should match previous snapshot', () => {
		const tree = renderer.create(
			<AudioOnlyMessage book={book} chapter={chapter} />,
		);

		expect(tree).toMatchSnapshot();
	});
	it('Should render a div containing given book and chapter', () => {
		const wrapper = Enzyme.shallow(<AudioOnlyMessage book={book} chapter={chapter} />);

		expect(
			wrapper
				.find('p')
				.text()
				.includes(book),
		).toBe(true);
		expect(
			wrapper
				.find('p')
				.text()
				.includes(chapter),
		).toBe(true);
	});
});
