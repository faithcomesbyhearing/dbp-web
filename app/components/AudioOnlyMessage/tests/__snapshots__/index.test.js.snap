// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioOnlyMessage Component Should match previous snapshot 1`] = `
<div
  className="audio-only"
>
  <svg>
    <use
      xlinkHref="/svglist.svg#audio_only"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
  </svg>
  <p>
    Romans 8
  </p>
  <span
    className="divider"
  />
  <span>
    Audio Only
  </span>
</div>
`;
